### ISSUE TITLE
bold, italic, bold-italic

### ISSUE THREAD

**Issue by blobject** [agarick/agave#5]
do it

**Comment by blobject** [comment]
Timeline is uncertain, unfortunately. Probably get to it in the summer but hopefully make steady additions over the next few months.

**Comment by aasutossh** [comment]
Please do the italics first.

**Comment by blobject** [comment]
Appreciate the suggestion. I'm quite busy currently but will start italic consideration once I have time.

**Comment by hwooo** [comment]
I hope you to consider bolded italics also.

**Comment by shabahengam** [comment]
Has there been any progress?

**Comment by blobject** [comment]
Hi @shabahengam. I've unfortunately been quite busy with personal matters, and will continue to be for several months, and haven't made progress with this. I'm still playing with ideas for italics. Bold seems like a simpler task.

**Comment by shabahengam** [comment]
ok,i wish you the best and i hope one day you find time to do it cause i really like this font.

**Comment by blobject** [comment]
Thank you! I really want to implement this as well, after the user feedback, and will do so as soon as I find time.

**Comment by blobject** [comment]
Bold has been implemented in https://github.com/blobject/agave/releases/tag/v35 (just ASCII so far). Please try it out!

Italic ASCII and BoldItalic ASCII coming soon(ish).

**Comment by shabahengam** [comment]
Wow, Thank you so much for working on this.
In my limited test just two letters feel a little weird.the letter `M` and `r`.please look at the picture to know what I'm talking about
![mr](https://user-images.githubusercontent.com/53096747/98550530-77d5fb00-22b1-11eb-8677-9a175fe6ba37.png)

**Comment by blobject** [comment]
Thanks for the feedback :D 

**summary of design**

Right, so for this first revision of bold, I kept the widths of glyphs the same as those of regular, while thickening the strokes, so you will notice that the space between strokes (the red span within "r" in your image, for example) get smaller.

I noticed this also has the effect of making the bold variant "feel smaller" when compared with the regular variant at the same size.

So I'm still unsure whether this decision is a good one, and might need to revisit it after more use and feedback.

**clarification**

Could you perhaps explain a bit more what you mean by "feel a little weird"? I know it can be hard to describe these feelings, and I assume you refer to the squeezed inner-spaces that I just mentioned, but maybe you are seeing something else.

**note on M**

The higher middle section of "M", on the other hand, was a more deliberate choice. I was worried keeping it level with the regular variant would make the region look too thick, especially at smaller sizes, and make it look like a messy blob, rather than a proper "M". Also, raising the middle helps to distinguish the glyph from "H".

**Comment by shabahengam** [comment]
> I assume you refer to the squeezed inner-spaces that I just mentioned

Yes,as I said before my English is not very good.sorry for confusing you :)

> The higher middle section of "M", on the other hand, was a more deliberate choice

OK.what about this parts? I mean they are not "straight" and have a little "curve"
![Untitled](https://user-images.githubusercontent.com/53096747/98563643-a8716100-22c0-11eb-8f7e-4c0508424b66.png)

**Comment by blobject** [comment]
Ah, I see. Perhaps the ink traps used in "M" there are a bit too dramatic. I'll look into it.

**Comment by zoomlogo** [comment]
What is the progress on this (and other issues)???

**Comment by blobject** [comment]
Stagnant. I've been too busy with other things; should get back to it in a few months.

**Comment by alexmyczko** [comment]
maybe the easiest would be:
In general, italics tend to slant between 4–14 degrees. Most contemporary fonts slant between 6–9 degrees.
try them all and pick the visually most appealing? being a mono font i would go for less than more…

**Comment by alexmyczko** [comment]
meanwhile, i've found 9 to be good:

https://sources.debian.org/src/fonts-agave/38-2/src/build

relevant part is Makefile that calls build. fontname+o = oblique version, i didn't manage to set font name right with the scripting:

https://fontforge.org/docs/scripting/scripting-alpha.html#SetFontNames

but I guess it would be easy to load the regular and bold sfd, do that editing in fontforge (skew 9) and save the sfd for releasing a next version?

well looking at the .sfd it's probably easier to just change it in the .sfd directly...

i can prepare a pull request if you like

**Comment by blobject** [comment]
Oh wow, I tried this out and it works well. Thank you very much for sharing. And the result actually looks pretty good, at least better than I can make it :P

And honestly, this will be a much faster and easier solution than trying to be cute and drawing everything by hand.

That said, I've noticed a few quirks in fontforge's skew:
- Some diacritics are misaligned. For example, "small letter o with caron" (u01d2)
- Some diacritics remain centered and require further adjustment to make them look correct. For example, "latin small letter e with acute" (u00e9)
- Some referenced glyphs skew the other way due to inversion. For example, "latin small capital inverted R" (u0281)
- A few referenced glyphs are off center, for some weird reason. For example, "latin capital letter upsilon" (u01b1)

Notes for myself:
- The skewing might not make sense for box-drawing and other graphical elements.

I think this is an excellent start, but I will at least have to solve the aforementioned problems before pushing a new version. Progress on this project has been slow, but this skew function will help things a lot. Thanks again for this, Alex!

**Comment by alexmyczko** [comment]
Just to let you know, people like your font! https://qa.debian.org/popcon-graph.php?packages=fonts-agave

**Comment by alexmyczko** [comment]
meanwhile fonts-agave in debian does build the oblique versions for regular and bold, however font systems fail to show them because i don't set the name right yet, any hints are welcome.
